### ISSUE TITLE
Static branching exercises error

### ISSUE THREAD

**Issue by sctyner** [wlandau/learndrake#35]
## Prework

- [X] Read and abide by `learndrake`'s [code of conduct](https://github.com/wlandau/learndrake/blob/master/CODE_OF_CONDUCT.md).
- [X] Search for duplicates among the [existing issues](https://github.com/wlandau/learndrake/issues), both open and closed.

## Description

The exercises page at https://colorado.rstudio.com/rsc/learndrake/static/ has an error. (4-static/4-static.Rmd)

## Reproducible example

Not applicable

## Session info

I got the error on Microsoft Edge Version 86.0.622.38 and Google Chrome Version 86.0.4240.75. Running Windows 10 Enterprise. 

## Expected output

A functioning `learnr` tutorial

**Comment by wlandau** [comment]
Thanks for catching that. Not sure what is wrong yet, but for now there is a working instance at https://wlandau.shinyapps.io/learndrakestatic/. Currently checking if I just need to regenerate `manifest.json`.

By the way, [`targets`](https://github.com/wlandau/targets) is the long-term successor to `drake`, and it is in the initial stages of [rOpenSci review](https://github.com/ropensci/software-review/issues/401). It solves several of `drake`'s permanent problems, including the reliance on a complicated DSL to declare each target. Static branching in [`targets`](https://github.com/wlandau/targets) is easier to learn, more consistent, and more reliable. The actual implementation is a [bunch of functions in `tarchetypes`](https://wlandau.github.io/tarchetypes/reference/index.html#section-branching), notably [`tar_map()`](https://wlandau.github.io/tarchetypes/reference/tar_map.html) and [`tar_combine()`](https://wlandau.github.io/tarchetypes/reference/tar_combine.html).

That said, `drake` is never going away, and sticking with it is a reasonable choice. More details are at https://wlandau.github.io/targets/articles/need.html#drake.

**Comment by wlandau** [comment]
@colearendt, do you happen to know what would explain [this parsing error](https://colorado.rstudio.com/rsc/learndrake/static/)? [`manifest.json`](https://github.com/wlandau/learndrake/blob/master/inst/tutorials/learndrakestatic/manifest.json) appears valid when I parse it locally with `jsonlite::fromJSON()`.

**Comment by colearendt** [comment]
@wlandau It looks like the issue is here: https://github.com/wlandau/learndrake/blob/1d7bd19d336f2b75d64364d18b8d455bad73370f/inst/tutorials/learndrakestatic/manifest.json#L1724 (I searched for that version number and then poked around that point in the file). Some libraries (i.e. `jsonlite::fromJSON()`) are a bit more forgiving on JSON parsing, but may just truncate part of the file. There may be a way to make it more strict though - I'm not sure off-hand.

Was this manifest produced directly by `rsconnect::writeManifest()`?

(Also, do you have a workshop coming up?)

**Comment by wlandau** [comment]
Yes, the manifest was generated by `rsconnect::writeManifest()` (with `rsconnect` 0.8.16). Removing the `lifecycle` package entry was worth a shot but does not seem to have worked.

I no longer have `drake` workshops coming up, but I did teach https://github.com/wlandau/targets-tutorial last week at R/Pharma. The RStudio Cloud space worked super well, and I plan to reuse it for subsequent iterations. the `targets` workshop is entirely notebook-based and does not have apps.

**Comment by colearendt** [comment]
Awesome! Good to hear! I believe this may be related:

```
curl: HTTP 404 https://colorado.rstudio.com/rspm/all/__linux__/bionic/latest/src/contrib/Archive/storr/storr_1.2.4.tar.gz
10/13 11:50:44.438 (GMT)
curl: (22) The requested URL returned error: 404
10/13 11:50:45.441 (GMT)
FAILED
10/13 11:50:45.442 (GMT)
Error in getSourceForPkgRecord(pkgRecord, srcDir(project), availablePackagesSource(repos = repos), : Failed to retrieve package sources for storr 1.2.4 from CRAN (internet connectivity issue?) [1.2.1 is current]
10/13 11:50:45.442 (GMT)
10/13 11:50:45.442 (GMT)
Unable to fully restore the R packages associated with this deployment.
10/13 11:50:45.442 (GMT)
Please review the preceding messages to determine which package
10/13 11:50:45.442 (GMT)
encountered installation difficulty and the cause of the failure.
10/13 11:50:45.452 (GMT)
Warning message:
10/13 11:50:45.452 (GMT)
In getSourceForPkgRecord(pkgRecord, srcDir(project), availablePackagesSource(repos = repos), :
10/13 11:50:45.452 (GMT)
Package version '1.2.4' is newer than the latest version reported by CRAN ('1.2.1') -- packrat may be unable to retrieve package sources.
```

The packrat restore actually failed for your most recent deployment, because you are using a newer version of the `storr` package than is available on RSPM. If it's possible to downgrade `storr`, that might fix the deployment issue. Otherwise, RSPM will likely update `storr` in a day or so and we can redeploy. It's curious that the roll-back failed with this strange issue.

**Comment by wlandau** [comment]
Thanks, Cole! Looks like it's working now.

**Comment by colearendt** [comment]
Just to be clear, it's running on the old version right now. I'll make a note to try redeploying the latest tomorrow when RSPM is updated.
